File: app/src/main/java/org/tvheadend/tvhclient/model/Connection.java
package org.tvheadend.tvhclient.model;

public class Connection {
    public long id = 0;
    public String name = "";
    public String address = "";
    public int port = 9982;
    public String username = "";
    public String password = "";
    public boolean selected = false;
    public int channelTag = 0;
    public int streaming_port = 9981;
    public String wol_mac_address = "";
    public int wol_port = 9;
    public boolean wol_broadcast = false;
    public int playback_profile_id = 0;
    public int recording_profile_id = 0;
    public int cast_profile_id = 0;
}

File: app/src/main/java/org/tvheadend/tvhclient/data/entity/Connection.kt
package org.tvheadend.tvhclient.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "connections", indices = [Index(value = ["id"], unique = true)])
data class Connection(

        @PrimaryKey(autoGenerate = true)
        var id: Int = 0,
        var name: String? = "",
        var hostname: String? = "",
        var port: Int = 9982,
        var username: String? = "",
        var password: String? = "",
        @ColumnInfo(name = "active")
        var isActive: Boolean = false,
        @ColumnInfo(name = "streaming_port")
        var streamingPort: Int = 9981,
        @ColumnInfo(name = "wol_enabled")
        var isWolEnabled: Boolean = false,
        @ColumnInfo(name = "wol_hostname")
        var wolMacAddress: String? = "",
        @ColumnInfo(name = "wol_port")
        var wolPort: Int = 9,
        @ColumnInfo(name = "wol_use_broadcast")
        var isWolUseBroadcast: Boolean = false,
        @ColumnInfo(name = "last_update")
        var lastUpdate: Long = 0,
        @ColumnInfo(name = "sync_required")
        var isSyncRequired: Boolean = true,
        @ColumnInfo(name = "server_url")
        var serverUrl: String? = "",
        @ColumnInfo(name = "streaming_url")
        var streamingUrl: String? = ""
)
